### ISSUE TITLE
[binary_ng] fused unary activations use the bf16 SFPU algorithms on float32 tensors (softplus returns 0 for x < -5)

### ISSUE THREAD

**Issue by sjettTT** [tenstorrent/tt-metal#56021]
### Component / Area

ops (ttnn eltwise binary_ng, SFPU)

### Issue Type

Bad Outputs

### Observed

On Blackhole, `ttnn.add(a, b, activations=[ttnn.UnaryWithParam(ttnn.UnaryOpType.SOFTPLUS, [1.0, 20.0])])` with
float32 inputs and output returns exactly 0 for every element where `x <= -5.02` (469 of 1024 probe points on a ramp
over `[-16, 8]`) and is about `1.5e-3` off elsewhere (`softplus(0) = 0.6942` against `ln 2 = 0.6931`). The standalone
`ttnn.softplus(x, beta=1.0, threshold=20.0)` on the same float32 tensor is never zero on that range and matches
`torch.nn.functional.softplus` to within the SFPU's float32 accuracy. The fused and standalone paths therefore give
different results for the same documented function on the same dtype.

Reading the code: the unary op's program factory defines `INP_FLOAT32` / `INP_INT32` / `INP_UINT32` / `INP_FLOAT`
from the input dtype (`ttnn/cpp/ttnn/operations/eltwise/unary/device/unary_program_factory.cpp:26-35, 419`). The
binary_ng factory composes fused activations through `add_activation_defines`
(`binary_ng/device/binary_ng_utils.cpp:581-600`, called from `binary_ng_program_factory.cpp:987-1012`), which emits
the init/func macros and `update_macro_defines` but no input-dtype define. The Blackhole SFPU softplus kernel
(`tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_softplus.h`) selects its algorithm on
`INP_FLOAT32`: the float32 branch uses a degree-8 residual polynomial plus an exp tail; the other branch uses a
degree-6 polynomial and, by design for bf16 precision, clamps the residual to 0 for `|beta * x| > 5` (header comment,
lines 26-31). With the define absent the fused path compiles the bf16 branch for float32 operands. `ckernel_sfpu_gelu.h`,
`ckernel_sfpu_erf.h`, `ckernel_sfpu_i1.h` and `ckernel_sfpu_digamma.h` key on the same define, so their fused
variants are affected the same way; softplus and erf were measured (see the first comment), gelu shows no difference on the probe.

Upstream float32 binary tests (`tests/ttnn/unit_tests/operations/eltwise/test_binary_fp32.py`,
`test_binaryng_fp32.py`) do not exercise a fused SOFTPLUS activation.

### Expected

A fused activation on a float32 tensor produces the same result as the standalone unary op on that tensor (the
float32 algorithm), within the SFPU's float32 accuracy; in particular no exact zeros for `softplus` at `x > -20`.

### 1. Steps (exact commands)

```python
x = torch.linspace(-16.0, 8.0, 1024, dtype=torch.float32).reshape(1, 1, 32, 32)
a = ttnn.from_torch(x, dtype=ttnn.float32, layout=ttnn.TILE_LAYOUT, device=device)
z = ttnn.zeros_like(a)
fused = ttnn.add(a, z, activations=[ttnn.UnaryWithParam(ttnn.UnaryOpType.SOFTPLUS, [1.0, 20.0])])
ref = ttnn.softplus(a, beta=1.0, threshold=20.0)
# compare ttnn.to_torch(fused), ttnn.to_torch(ref), torch.nn.functional.softplus(x)
```

### 1. Software Versions

tt-metal `d04395ed86` (2026-08-29). `binary_ng_utils.cpp`, `unary_op_utils.cpp` and `ckernel_sfpu_softplus.h` are unchanged on `main` `89139764a0e` (2026-09-09); the `main` changes to `binary_ng_program_factory.cpp` and `unary_program_factory.cpp` since then do not touch activation or dtype defines. tt-kmd 2.8.0, Python 3.10.12.

### 2. Hardware Details

Blackhole p150. Wormhole not tested.

### Is this a regression?

Unknown.

### Logs & Diagnostics

Not attached; the probe above reproduces it in a few seconds.

**Comment by sjettTT** [comment]
Closing until this is reproduced in isolation on hardware; will reopen with the probe results if it holds.

**Comment by sjettTT** [comment]
Reproduced in isolation on Blackhole p150 (tt-metal main `9fe0ba04fc`), probe: `ttnn.add(x, 0, activations=[act])` vs the standalone unary op vs torch on a 1024-point float32 ramp over [-16, 8].

| fp32, fused in `ttnn.add` | exact zeros | max abs err vs torch | standalone op: zeros / max err | fused vs standalone max diff |
| --- | --- | --- | --- | --- |
| SOFTPLUS(1, 20) | 469 / 1024 (x <= -5.021) | 6.6e-3 (at x = 5.02) | 0 / 4.2e-5 | 6.6e-3 |
| ERF | 0 | 3.2e-3 | 0 / 2.4e-7 | 3.2e-3 |
| GELU | 446 | 9.5e-7 | 446 / 9.5e-7 | 0 |

On bf16 every fused result is bit-identical to the standalone op (both return the bf16 algorithm's values, e.g. softplus is 0 below -5 by design there). With the input-dtype defines supplied to the fused path, all three fp32 fused results become bit-identical to their standalone ops (0 zeros, 4.2e-5 / 2.4e-7 / 9.5e-7).
